File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/surveys/core/outbound/jpa/SurveyRepository.java
package io.zenwave360.example.clinicaltool.modules.surveys.core.outbound.jpa;

import io.zenwave360.example.clinicaltool.modules.surveys.core.domain.*;
import java.math.*;
import java.time.*;
import java.util.*;

import org.springframework.data.jpa.repository.*;
import org.springframework.stereotype.Repository;

/**
 * Spring Data JPA repository for the Survey entity.
 */
@SuppressWarnings("unused")
@Repository
public interface SurveyRepository extends JpaRepository<Survey, Long> {java.util.Optional<Survey> findByNameAndHospitalId(String name, Long hospitalId);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/surveys/repository/jpa/SurveyRepository.kt
package io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*
import java.math.*
import java.time.*
import java.util.*

import org.springframework.data.jpa.repository.*
import org.springframework.stereotype.Repository

/**
 * Spring Data JPA repository for the Survey entity.
 */
@Suppress("unused")
@Repository
interface SurveyRepository : JpaRepository<Survey, Long> {fun findByNameAndHospitalId(name: String, hospitalId: Long): java.util.Optional<Survey>
}
